packages/checkout-storefront/src/sections/Addresses/useAddressAvailability.ts

```diff
@@ -0,0 +1,27 @@
+import { CountryCode, useChannelQuery } from "@/checkout-storefront/graphql";
+
+interface UseAddressAvailabilityProps {
+  pause: boolean;
+}
+
+export const useAddressAvailability = (
+  { pause }: UseAddressAvailabilityProps = { pause: false }
+) => {
+  const [{ data }] = useChannelQuery({
+    variables: { slug: "default-channel" },
+    pause,
+  });
+
+  const availableShippingCountries: CountryCode[] =
+    (data?.channel?.countries?.map(({ code }) => code) as CountryCode[]) || [];
+
+  const isAvailable = ({ country }: { country: { code: string } }) => {
```

Comment:
I'd love to buuuut currently in the api addresses have this as string and in this function I'm actually providing pieces of addresses - there is a ticket for that but since it's a breaking change and could ruin ppls types in their apps we can only fix it along Saleor 4.0 so gotta wait for that and then we can change to CountryCode ✨